pyannote/database/file_finder.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+# The MIT License (MIT)
+
+# Copyright (c) 2016- CNRS
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# AUTHORS
+# Hervé BREDIN - http://herve.niderb.fr
+# Alexis PLAQUET
+
+import warnings
+from pathlib import Path
+from typing import Text
+from pyannote.database.protocol.protocol import ProtocolFile
+from .registry import registry as global_registry
+from .registry import Registry
+
+
+class FileFinder:
+    """Database file finder. 
+    
+    Retrieve media files by URI.
+
+    Parameters
+    ----------
+    registry : Registry, optional
+        Database registry. Defaults to `pyannote.database.registry`.
+    """
+
+    def __init__(
+        self, 
+        registry: Registry = None,
+        database_yml: Text = None):
+        super().__init__()
+        if registry is None:
+            if database_yml is None:
+                registry = global_registry
+            else:
+                warnings.warn("Passing `database.yml` to `FileFinder` is deprecated in favor of `registry`.")
+                registry = Registry()
+                registry.load_database(database_yml)
```

Comment:
@FrenchKrab the missing `s` at the end of `load_database` is done on purpose (as I plan on pushing some more commits that remove the option to load multiple YAML files at once.